src/plots/plots.js

```diff
@@ -454,6 +454,9 @@ plots.supplyDefaults = function(gd) {
 
     newFullLayout._initialAutoSizeIsDone = true;
 
+    // pass along trace colors
+    newFullLayout._colorway = newLayout.colorway;
```

Comment:
That works, but we can do better. Ideally, we should use [`Lib.coerce`](https://github.com/plotly/plotly.js/blob/master/src/lib/coerce.js) to validate and map `colorway` from input layout to full layout. `layout.colorway` is a bit of a special case: we don't have a `valType` that corresponds to the desired behavior. That said, we should at the very least make sure that colorway is an array and (manually) validate every color in the `colorway`.